### PR TITLE
Improve Security

### PR DESCRIPTION
**Pull request by DRvanR**
Based on https://www.pivotaltracker.com/story/show/137540029

will require a major version bump due to removed support for PHP < 5.6

### CODE REVIEW

**Comment by DRvanR** [comment]
@arothuis

File: build.xml [hunk 1]
```diff
@@ -1,8 +1,10 @@
 <?xml version="1.0" encoding="UTF-8"?>
 
 <project name="Yubikey API Client-ci" default="build">
-        <target name="build"
-            depends="php-lint-ci,phpmd-ci,phpcs-ci,phpcpd-ci,phpunit-ci"/>
+        <target name="build" depends="php-lint,phpmd,phpcs,phpcpd,alltests,security-checker"/>
+        <target name="pre-commit" depends="php-lint,phpmd,phpcs,phpcpd,unittests"/>
+        <target name="pre-push" depends="phpmd,phpcs,phpcpd,alltests"/>
+        <target name="nightly" depends="integrationtests,security-checker"/>
```
Comment (arothuis) [inline build.xml]: :bed: Nice.

File: build.xml [hunk 2]
```diff
@@ -65,17 +67,36 @@
...
 
-    <target name="phpcpd-ci" description="Find duplicate code using PHPCPD">
+    <target name="phpcpd" description="Find duplicate code using PHPCPD">
         <exec executable="vendor/bin/phpcpd" failonerror="false">
             <arg path="src"/>
         </exec>
     </target>
 
-    <target name="phpunit-ci" description="Run unit tests with PHPUnit">
+    <target name="alltests" description="Run the unit and integration tests" depends="unittests,integrationtests">
```
Comment (arothuis) [inline build.xml]: Nit: I'd rather see this and other relevant targets separated by a dash, e.g. `all-tests`, `integration-tests`, `unit-tests`...

**Review by arothuis — APPROVED** [review]
💯, some nits (not important); thanks for splitting up the test suites

**Comment by arothuis** [comment]
Thanks! Feel free to merge away 👍

**Pull request merged**